src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/activities/ActivitiesResourceIntTest.kt

```diff
@@ -179,91 +179,192 @@ class ActivitiesResourceIntTest : IntegrationTestBase() {
+      }
+
+      @Test
+      fun `access forbidden when no role`() {
+        webTestClient.put().uri("/activities/1")
+          .contentType(MediaType.APPLICATION_JSON)
+          .headers(setAuthorisation(roles = listOf()))
+          .body(BodyInserters.fromValue(updateActivityRequestJson()))
+          .exchange()
+          .expectStatus().isForbidden
+      }
+
+      @Test
+      fun `access forbidden with wrong role`() {
+        webTestClient.put().uri("/activities/1")
+          .contentType(MediaType.APPLICATION_JSON)
+          .headers(setAuthorisation(roles = listOf("ROLE_BANANAS")))
+          .body(BodyInserters.fromValue(updateActivityRequestJson()))
+          .exchange()
+          .expectStatus().isForbidden
+      }
+
+      @Test
+      fun `should call the service`() {
+        doReturn(UpdateActivityResponse(prisonId = "LEI")).whenever(activityService).updateActivity(anyLong(), any())
+
+        webTestClient.put().uri("/activities/1")
+          .contentType(MediaType.APPLICATION_JSON)
+          .headers(setAuthorisation(roles = listOf("ROLE_NOMIS_ACTIVITIES")))
+          .body(BodyInserters.fromValue(updateActivityRequestJson()))
+          .exchange()
+          .expectStatus().isOk
+
+        verify(activityService).updateActivity(1, updateActivityRequest())
+      }
+
+      @Test
+      fun `should return bad request`() {
+        doThrow(BadDataException("Prison not found")).whenever(activityService).updateActivity(anyLong(), any())
+
+        webTestClient.put().uri("/activities/1")
+          .contentType(MediaType.APPLICATION_JSON)
+          .headers(setAuthorisation(roles = listOf("ROLE_NOMIS_ACTIVITIES")))
+          .body(BodyInserters.fromValue(updateActivityRequestJson()))
+          .exchange()
+          .expectStatus().isBadRequest
+      }
+
+      @Test
+      fun `should return not found`() {
+        doThrow(NotFoundException("Activity not found")).whenever(activityService).updateActivity(anyLong(), any())
+
+        webTestClient.put().uri("/activities/1")
+          .contentType(MediaType.APPLICATION_JSON)
+          .headers(setAuthorisation(roles = listOf("ROLE_NOMIS_ACTIVITIES")))
+          .body(BodyInserters.fromValue(updateActivityRequestJson()))
+          .exchange()
+          .expectStatus().isNotFound
+      }
+    }
 
     @Test
-    fun `access forbidden when no authority`() {
+    fun `should return bad request for missing data`() {
       webTestClient.put().uri("/activities/1")
         .contentType(MediaType.APPLICATION_JSON)
-        .body(BodyInserters.fromValue(updateActivityRequest()))
+        .headers(setAuthorisation(roles = listOf("ROLE_NOMIS_ACTIVITIES")))
+        .body(BodyInserters.fromValue(updateActivityRequestJson(prisonIdJson = null)))
         .exchange()
-        .expectStatus().isUnauthorized
+        .expectStatus().isBadRequest
+        .expectBody().jsonPath("$.userMessage").value<String> { it.contains("prisonId") }
```

Comment:
I think this would be more readable if you just assert on the entire message; it should be readable like "missing field prisonId" or whatever and there is no reason for it to change

Reply:
Actually this is the message:
`JSON parse error: Instantiation of [simple type, class uk.gov.justice.digital.hmpps.nomisprisonerapi.activities.UpdateActivityRequest] value failed for JSON property prisonId due to missing (therefore NULL) value for creator parameter prisonId which is a non-nullable type`
I'm not sure that's going to make the test clearer!